tests/unit/test_space.py

```diff
@@ -302,12 +302,65 @@ def test_box_contains_raises_on_point_of_different_shape(
 
 
 @pytest.mark.parametrize("num_samples", [0, 1, 10])
-def test_box_sampling(num_samples: int) -> None:
+def test_box_sampling_returns_correct_shape(num_samples: int) -> None:
     box = Box(tf.zeros((3,)), tf.ones((3,)))
     samples = box.sample(num_samples)
     _assert_correct_number_of_unique_constrained_samples(num_samples, box, samples)
 
 
+@pytest.mark.parametrize("num_samples", [0, 1, 10])
+def test_box_sobol_sampling_returns_correct_shape(num_samples: int) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    sobol_samples = box.sample_sobol(num_samples)
+    _assert_correct_number_of_unique_constrained_samples(num_samples, box, sobol_samples)
+
+
+@pytest.mark.parametrize("num_samples", [0, 1, 10])
+def test_box_halton_sampling_returns_correct_shape(num_samples: int) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    halton_samples = box.sample_halton(num_samples)
+    _assert_correct_number_of_unique_constrained_samples(num_samples, box, halton_samples)
+
+
+@pytest.mark.parametrize("num_samples", [-1, -10])
+def test_box_sampling_raises_for_invalid_sample_size(num_samples: int) -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        box = Box(tf.zeros((3,)), tf.ones((3,)))
+        box.sample(num_samples)
+
+
+@pytest.mark.parametrize("num_samples", [-1, -10])
+def test_box_sobol_sampling_raises_for_invalid_sample_size(num_samples: int) -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        box = Box(tf.zeros((3,)), tf.ones((3,)))
+        box.sample_sobol(num_samples)
+
+
+@pytest.mark.parametrize("num_samples", [-1, -10])
+def test_box_halton_sampling_raises_for_invalid_sample_size(num_samples: int) -> None:
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        box = Box(tf.zeros((3,)), tf.ones((3,)))
+        box.sample_halton(num_samples)
+
+
+@pytest.mark.parametrize("skip", [1, 10, 100])
+def test_box_sobol_sampling_returns_same_points_for_same_skip(skip: int) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    sobol_samples_1 = box.sample_sobol(num_samples=100, skip=skip)
+    sobol_samples_2 = box.sample_sobol(num_samples=100, skip=skip)
+    npt.assert_allclose(sobol_samples_1, sobol_samples_2)
+
+
+@pytest.mark.parametrize("seed", [1, 42, 123])
+def test_box_halton_sampling_returns_same_points_for_same_seed(seed: int) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    tf.random.set_seed(1)
+    halton_samples_1 = box.sample_halton(num_samples=100, seed=seed)
+    tf.random.set_seed(1)
```

Comment:
We would want this seed setting to happen in the sampler itself, so that it is reproducible. What do you think @uri-granta ?

Reply:
See discussion above for context

Reply:
@satrialoka  please add tf.random.set_seed(seed) into the .sample_halton method

Reply:
We want to be able to call .sample_halton(num_samples=100, seed=seed) and get exactly the same back

Reply:
okay